app/controllers/exam_templates_controller.rb

```diff
@@ -23,15 +26,14 @@ def update
     new_uploaded_io = params[:exam_template][:new_template]
     # error checking when new_uploaded_io is not pdf
     if new_uploaded_io.content_type != 'application/pdf'
-      flash_message(:error, t('exam_templates.update.failure'))
+      flash[:error] = t('exam_templates.update.failure')
       redirect_to action: 'index'
       return
     end
     assignment = Assignment.find(params[:assignment_id])
     old_exam_template = assignment.exam_templates.find_by(id: params[:id])
     old_template_filename = old_exam_template.filename
     old_exam_template.replace_with_file(new_uploaded_io.read, assignment_id: assignment.id, filename: old_template_filename)
-    flash_message(:success, t('exam_templates.update.success'))
-    redirect_to action: 'index'
+    respond_with(old_exam_template, :location => assignment_exam_templates_url)
```

Comment:
Never use `=>`. Always use the new hash syntax `location: ...`. I do not want to remind you of this again.